pandas/tests/frame/test_query_eval.py

```diff
@@ -695,6 +695,18 @@ def test_inf(self):
             result = df.query(q, engine=self.engine, parser=self.parser)
             tm.assert_frame_equal(result, expected)
 
+    def test_check_tz_aware_index_query(self, tz_aware_fixture):
+        # https://github.com/pandas-dev/pandas/issues/29463
+        tz = tz_aware_fixture
+        df_index = pd.date_range(
+            start="2019-01-01", freq="1d", periods=10, tz=tz, name="time"
+        )
+        expected = pd.DataFrame(index=df_index)
+
+        df = pd.DataFrame(index=df_index)
+        result = df.query('"2018-01-03 00:00:00+00" < time')
+        tm.assert_frame_equal(result, expected)
+
```

Comment:
can you also test the working case here, e.g. ```result = df.reset_index().query(....)```

Reply:
@jreback  Done. Please review.

Reply:
@vipinkjonwal thanks for your updates

Like this, what's the difference between `df2.set_index("time")` and `df`?

I think Jeff meant
```python
result = df.reset_index().query('"2018-01-03 00:00:00+00" < time')
expected = pd.DataFrame(df_index)
tm.assert_frame_equal(result, expected)
```

Reply:
@MarcoGorelli , I thought this was already working when the related issue was raised , I've updated the test with suggested changes.